src/openfhe.jl

```diff
@@ -381,3 +385,278 @@ function rotate(sv::SecureVector{<:OpenFHEBackend}, shift; wrap_by)
 
     secure_vector
 end
+
+
+############################################################################################
+# Matrix
+############################################################################################
+
+function init_matrix_rotation!(context::SecureContext{<:OpenFHEBackend}, private_key::PrivateKey,
+                               shifts_2d::Vector{Tuple{Int, Int}}, shape::Tuple{Int, Int})
+    cc = get_crypto_context(context)
+    shifts_1d = []
+    for (i, j) in shifts_2d
+        # minimum required shift
+        i %= shape[1]
+        j %= shape[2] 
+        # appropriate shift for matrix packed in vector
+        shift = []
+        if j == 0
+            if i != 0
+                shift = [i*shape[2]]
+            end
+        else
+            shift = [j+i*shape[2], -sign(j)*(shape[2]-abs(j))+i*shape[2]]
+        end
+        append!(shifts_1d, shift)
+        # additional shift required in case of rotation in i direction
+        if i != 0
+            shift_rest = -sign.(shift) .* (shape[1]*shape[2] .- abs.(shift))
+            append!(shifts_1d, shift_rest)
+        end
+    end
+    OpenFHE.EvalRotateKeyGen(cc, private_key.private_key, -shifts_1d)
+
+    nothing
+end
+
+function PlainMatrix(data::Vector{Float64}, context::SecureContext{<:OpenFHEBackend},
+                     shape::Tuple{Int, Int})
+    cc = get_crypto_context(context)
+    plaintext = OpenFHE.MakeCKKSPackedPlaintext(cc, data)
+    capacity = OpenFHE.GetSlots(plaintext)
+    plain_matrix = PlainMatrix(plaintext, shape, capacity, context)
+
+    plain_matrix
+end
+
+function PlainMatrix(data::Vector{<:Real}, context::SecureContext{<:OpenFHEBackend},
+                     shape::Tuple{Int, Int})
+    PlainMatrix(Vector{Float64}(data), context, shape)
+end
+
+function PlainMatrix(data::Matrix{<:Real}, context::SecureContext{<:OpenFHEBackend})
+    PlainMatrix(Vector{Float64}(vec(transpose(data))), context, size(data))
```

Comment:
Yeah, `vec` functions saved the matrix in a vector column-wise, all algorithms I wrote were row-wise. I think it is more logical because in memory normally (in C++ for example) consequent saved are row elements. May be in Julia it is not the case...

julia```
julia> m = [1 2 3; 4 5 6; 7 8 9]
3×3 Matrix{Int64}:
 1  2  3
 4  5  6
 7  8  9

julia> vec(m)
9-element Vector{Int64}:
 1
 4
 7
 2
 ⋮
 3
 6
 9

julia> vec(transpose(m))
9-element reshape(transpose(::Matrix{Int64}), 9) with eltype Int64:
 1
 2
 3
 4
 ⋮
 7
 8
 9
```

Reply:
Do you think I should rewrite algorithms in column-majored order? Probably it is only `rotate` function